src/transforms/kubernetes_pod_metadata.rs

```diff
@@ -0,0 +1,767 @@
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    fn transform_type(&self) -> &'static str {
+        "kubernetes_pod_metadata"
+    }
+}
+
+fn node_name() -> Result<String, BuildError> {
+    std::env::var(NODE_NAME_ENV).map_err(|_| BuildError::MissingNodeName { env: NODE_NAME_ENV })
+}
+
+fn kubernetes_host() -> Result<String, BuildError> {
+    std::env::var(HOST_ENV).map_err(|_| BuildError::NoKubernetes {
+        reason: format!("Missing Kubernetes API host defined with {}", HOST_ENV),
+    })
+}
+
+fn kubernetes_port() -> Result<String, BuildError> {
+    std::env::var(PORT_ENV).map_err(|_| BuildError::NoKubernetes {
+        reason: format!("Missing Kubernetes API port defined with {}", PORT_ENV),
+    })
+}
+
+fn account_token() -> Option<String> {
+    fs::read(TOKEN_PATH)
+        .map_err(|error| {
+            warn!(
+                message = "Missing Kubernetes service account token file.",
+                ?error
+            )
+        })
+        .ok()
+        .and_then(|bytes| {
+            String::from_utf8(bytes)
+                .map_err(|error| {
+                    warn!(
+                        message = "Kubernetes service account token file is not a valid utf8.",
+                        ?error
+                    )
+                })
+                .ok()
+        })
+}
+
+fn tls_settings() -> Result<TlsSettings, BuildError> {
+    let mut options = TlsOptions::default();
+    options.ca_path = Some(CA_PATH.into());
+    TlsSettings::from_options(&Some(options)).context(TlsError)
+}
+
+#[derive(Debug, Snafu)]
+enum BuildError {
+    #[snafu(display("{}, probably because Vector isn't in Kubernetes Pod.", reason))]
+    NoKubernetes { reason: String },
+    #[snafu(display("TLS construction errored {}.", source))]
+    TlsError { source: crate::Error },
+    #[snafu(display("Http client construction errored {}.", source))]
+    HttpError { source: crate::Error },
+    #[snafu(display(
+        "Missing environment variable {:?} containing node name `spec.nodeName`.",
+        env
+    ))]
+    MissingNodeName { env: &'static str },
+    #[snafu(display("Failed constructing request: {}.", source))]
+    K8SOpenapiError { source: k8s::RequestError },
+    #[snafu(display("Uri gotten from Kubernetes is invalid: {}.", source))]
+    InvalidUri { source: uri::InvalidUri },
+    #[snafu(display("Uri gotten from Kubernetes is invalid: {}.", source))]
+    InvalidUriParts { source: uri::InvalidUriParts },
+    #[snafu(display("Authorization token gotten from Kubernetes is invalid: {}.", source))]
+    InvalidToken { source: header::InvalidHeaderValue },
+}
+
+/// Background client which watches for Pod metadata changes and propagates them to Transform.
+struct MetadataClient {
+    fields: Vec<Box<dyn Fn(&Pod) -> Vec<(Atom, Value)> + Send + Sync + 'static>>,
+    sender: Option<mpsc::Sender<(Bytes, Vec<(Atom, Value)>)>>,
```

Comment:
Why a channel over using `evmap`?

Reply:
Didn't want to depend on extra dependencies. 

But as we are already depending on `evmap`, it's a better option, so I'll change it to that.